### PR TITLE
Added root_notation flag to latex and pretty printers

### LINKED ISSUE #14033: Add a flag root_notation to relevant printers
Some printers, at least "latex" and "pretty", use root notation for certain fractional powers:
```
>>> pprint(x**(S(1)/3), use_unicode=False)
3 ___
\/ x
>>> latex(x**(S(1)/3))
'\\sqrt[3]{x}'
```
But this is not always desired. There should be a flag `root_notation`, True by default, setting which to False will cause the printers to not use that notation, printing fractional powers instead. 

The relevant method of each printer is `_print_Pow`, see [LaTeX for example](https://github.com/sympy/sympy/blob/master/sympy/printing/latex.py#L482). The part with "Treat x**Rational(1,n) as special case" would not be entered if `root_notation` is False.

### PR DESCRIPTION
**Pull request by prajwalgatti**
Fixes #14033

This PR fixes the issue of not being able to use fractional form for powers with exponents of the form 1/n in LaTex and pretty printers. 

`root_notation` flag is added to [LaTeX](http://docs.sympy.org/dev/_modules/sympy/printing/latex.html)  and [pretty](http://docs.sympy.org/dev/_modules/sympy/printing/pretty/pretty.html) printers, which specifies whether the above-mentioned expressions are to be printed in their root form or fractional form. 

The default value of `root_notation` in both printers is set to `True`, which prints the root form of the expression.

Example:

In LaTex
```
>>> print(latex(x**Rational(1, 7)))
\sqrt[7]{x}

>>> print(latex(x**Rational(1, 7),  root_notation = False))
x^{\frac{1}{7}}

```
In prettyprinting

```
>>> print(pretty(x**Rational(1, 5), use_unicode = False))
5 ___
\/ x 

>>> print(pretty(x**Rational(1, 5), use_unicode = False, root_notation = False))
 1/5
x   

```

### CODE REVIEW

**Review by normalhuman — COMMENTED** [review]
Thanks, but there are five more things to do:

1. Add a test for this flag in latex printer to https://github.com/sympy/sympy/blob/master/sympy/printing/tests/test_latex.py 
2. Add a test for this flag in pretty printer to https://github.com/sympy/sympy/blob/master/sympy/printing/pretty/tests/test_pretty.py
3. Add a description of this flag to the docstring of latex function, see http://docs.sympy.org/dev/_modules/sympy/printing/latex.html#latex
4. Add a description of this flag to the docstring of pretty_print function, see http://docs.sympy.org/dev/_modules/sympy/printing/pretty/pretty.html#pretty_print
5. In the text of the pull request, use the phrase "Fixes #xxxxx" or "Closes #xxxxx" so that the referenced issue is closed automatically when the PR is merged. "Reference to" will not do that. 

Also, see the recommendations on [writing commit messages](https://github.com/sympy/sympy/wiki/Development-workflow#writing-commit-messages) from Development Workflow. Someone may be bisecting the commit tree a year later trying to debug a problem; having informative commit messages helps.

**Comment by prajwalgatti** [comment]
@normalhuman I have made the necessary changes 
Is there anything else I should have done?
Thank you for guiding

File: sympy/printing/pretty/pretty.py [hunk 1]
```diff
@@ -2284,6 +2285,10 @@ def pretty_print(expr, **settings):
     use_unicode : bool or None, optional
         use unicode characters, such as the Greek letter pi instead of
         the string pi.
+    root_notation : bool, optional
+        use root form or fractional form for expressions having rational
+        roots of the form 1/n. Set to False for fractional form;
```
Comment (normalhuman) [inline sympy/printing/pretty/pretty.py]: "expressions having rational roots" means something different. Also, "or" is typically not used when describing a boolean parameter; for example, "use_unicode" says "use unicode characters, such as the Greek letter pi". It is understood that False means: don't do that thing.

I suggest "use root form for powers with exponent of the form 1/n. Default is True."

**Comment by asmeurer** [comment]
Perhaps we should use True/False/None, where None is the current behavior, and True forces the root notation for all fractional powers. Right now it only uses root notation if the power is a fraction and the numerator is 1.

**Comment by asmeurer** [comment]
Actually if you look at `x**(1/y)` there is currently a discrepancy between the LaTeX and pretty printers, which should be fixed.

**Comment by shiksha11** [comment]
@prajwalgatti I am interested in fixing this issue. Can I take your work so far and continue with it??

**Comment by prajwalgatti** [comment]
> @prajwalgatti I am interested in fixing this issue. Can I take your work so far and continue with it??

Go ahead

**Comment by oscargus** [comment]
Superseded by #15842

**Pull request closed without merging**